core/src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/AttributeComponentSpec.scala

```diff
@@ -26,47 +26,49 @@ class AttributeComponentSpec extends TestDriverComponentWithFlatSpecAndMatchers
     records should contain theSameElementsAs { expectedRecords map { _.copy(id = dummyId1) }}
   }
 
+  def insertWorkspaceAttributeRecords(workspaceId: UUID, attributeName: AttributeName, attribute: Attribute): ReadWriteAction[Option[Int]] = {
+    entityQuery.findEntityByWorkspace(workspaceId).result flatMap { entityRecords =>
+      val entityIdsByRef = entityRecords.map(rec => AttributeEntityReference(rec.entityType, rec.name) -> rec.id).toMap
+      workspaceAttributeQuery ++= workspaceAttributeQuery.marshalAttribute(workspaceId, attributeName, attribute, entityIdsByRef)
+    }
+  }
+
   "AttributeComponent" should "insert string attribute" in withEmptyTestDatabase {
     val testAttribute = AttributeString("test")
     runAndWait(workspaceQuery.save(workspace))
-    val numRows = workspaceAttributeQuery.insertAttributeRecords(workspaceId, AttributeName.withDefaultNS("test"), testAttribute, workspaceId).map(x => runAndWait(x))
```

Comment:
The `numRows` tests are redundant since we're already checking that records were inserted correctly.  I was unable to replicate the odd behavior of `Seq(1, 1, 1, 1)` meaning "inserted 4 rows" and I don't think it's valuable to spend any more effort on that.